src/CTA.Rules.Metrics/PortSolutionResultReportGenerator.cs

```diff
@@ -52,24 +76,34 @@ private void GenerateMetrics()
             var targetVersionMetrics = new List<TargetVersionMetric>();
             var upgradePackageMetrics = new List<UpgradePackageMetric>();
             var actionPackageMetrics = new List<ActionPackageMetric>();
-            var actionMetrics = new List<GenericActionMetric>();
             var actionExecutionMetrics = new List<GenericActionExecutionMetric>();
 
             foreach (var projectResult in PortSolutionResult.ProjectResults)
             {
                 targetVersionMetrics.AddRange(MetricsTransformer.TransformTargetVersions(Context, projectResult));
                 upgradePackageMetrics.AddRange(MetricsTransformer.TransformUpgradePackages(Context, projectResult));
                 actionPackageMetrics.AddRange(MetricsTransformer.TransformActionPackages(Context, projectResult));
-                actionMetrics.AddRange(MetricsTransformer.TransformProjectActions(Context, projectResult));
                 actionExecutionMetrics.AddRange(MetricsTransformer.TransformGenericActionExecutions(Context, projectResult));
             }
 
             TargetVersionMetrics = targetVersionMetrics;
             UpgradePackageMetrics = upgradePackageMetrics;
             ActionPackageMetrics = actionPackageMetrics;
-            GenericActionMetrics = actionMetrics;
             GenericActionExecutionMetrics = actionExecutionMetrics;
-        } 
+        }
+
+        private string GenerateAnalyzeSolutionResultJsonReport()
+        {
+            var allMetrics = new List<CTAMetric>();
+            allMetrics = allMetrics
+                .Concat(UpgradePackageMetrics)
+                .Concat(ActionPackageMetrics)
+                .Concat(GenericActionMetrics)
+                .ToList();
+
+            PortSolutionResultJsonReport = JsonConvert.SerializeObject(allMetrics);
+            return PortSolutionResultJsonReport;
+        }
```

Comment:
Do you think there is any risk sharing the `PortSolutionResultJsonReport` property with the other report-generating method `GeneratePortSolutionResultJsonReport()`?

This also might be an indicator that this class should be refactored such that we have static methods to generate and export the different reports. If you believe that's the case, then we can save the refactor for another issue + PR.